rfc/datawherehouse.md

```diff
@@ -0,0 +1,132 @@
+# DATAWHEREHOUSE
+
+> or... where is that file living on?
+
+## Authors
+
+- [Vasco Santos], [Protocol Labs]
+
+## Background
+
+TL;DR
+
+1. We're missing a mapping of Data CID -> target URI (Bucket, Saturn Node, etc).
+2. We don't want clients to create location claims for internal bucket URLs because we might change the location in the future (reputation hit for client).
+3. We don't have bucket events in Cloudflare, so need the client to tell us when it has uploaded something to the provided write target.
+4. We want freeway code to be usable in Saturn nodes, so ideally it uses only content claims to discover locations.
+5. We want this information available as soon as content is written so that read interfaces can serve the content right away.
```

Comment:
Putting this comment here as I'm not sure what better place to put this would be. This is followup from the call this morning. I think we should issue a location claim when we verify content was delivered. Challenge however was how do we capture information like things like `bucket`, `node` etc.. not present in the URL so we could perform compute closer to a data.

I think there are few possible solution here:

1. We encode all that information as hints e.g. `https://w3s.link/car/bag...car?r2&bucket=foo`
    - 💚 Anyone will be able to perfrom read from that gateway URL
    - 💚 Gateway could take advantage of those hints or not care about them.
    - 💚 Other things in the system can recognize those hints and perform direct from bucket read as opposed to just going to that URL (backdoor / fastlane access)

2. We could publish claim with gateway URL and publish bucket and other info as a separate (non-location) claim
    - 💚 Anyone will be able to perform read from that gateway URL
    - 💚 Internal system can query both claims and to device most effecient access

3. We could publish claims with our custom URL schemas e.g. r2://bucket/.... 
    - ❤️‍🩹 Anyone who knows how to resolve it can read it, we can even publish resolver library
       - Our gateway could even support `https://w3s.link/r2://bucket/.../cid` URL
    - 💚 Internal system can query both claims and to device most efficient access

Broadly speaking I'm advocating for capturing whatever extra information we need to enable more optimal reads for internal use, yet make URLs publicly readable from the outside perhaps in less optimal mannery.

Reply:
Thanks for the write up. I now understand what you mean. The "Gateway could take advantage of those hints or not care about them" doing whatever other discovery they feel like/know about makes sense to me.

Yes, I think this a good suggestion